server/src/test/java/org/elasticsearch/cluster/service/MasterServiceTests.java

```diff
@@ -982,6 +983,127 @@ public void onAckTimeout() {
         }
     }
 
+    @TestLogging(value = "org.elasticsearch.cluster.service.MasterService:WARN", reason = "testing WARN logging")
+    public void testStarvationLogging() throws Exception {
+        final long warnThresholdMillis = MasterService.MASTER_SERVICE_STARVATION_LOGGING_THRESHOLD_SETTING.get(Settings.EMPTY).millis();
+        relativeTimeInMillis = randomLongBetween(0, Long.MAX_VALUE - warnThresholdMillis * 3);
+        final long startTimeMillis = relativeTimeInMillis;
+        final long taskDurationMillis = TimeValue.timeValueSeconds(1).millis();
+
+        MockLogAppender mockAppender = new MockLogAppender();
+        mockAppender.start();
+
+        Logger clusterLogger = LogManager.getLogger(MasterService.class);
+        Loggers.addAppender(clusterLogger, mockAppender);
+        try (MasterService masterService = createMasterService(true)) {
+            final AtomicBoolean keepRunning = new AtomicBoolean(true);
+
+            final Runnable await = new Runnable() {
+                private final CyclicBarrier cyclicBarrier = new CyclicBarrier(2);
+
+                @Override
+                public void run() {
+                    try {
+                        cyclicBarrier.await(5, TimeUnit.SECONDS);
+                    } catch (InterruptedException | BrokenBarrierException | TimeoutException e) {
+                        throw new AssertionError("unexpected", e);
+                    }
+                }
+            };
+            final Runnable awaitNextTask = () -> {
+                await.run();
+                await.run();
+            };
+
+            final ClusterStateUpdateTask starvingTask = new ClusterStateUpdateTask(Priority.HIGH) {
```

Comment:
`starving` made me thing this task is starving. I wonder if `greedy` was better to signal that it starves out the other task. It is probably correct English and possibly just me not understanding the nuances.

Reply:
I see the confusion :grin: Both "starved" and "starving" can mean "suffering from starvation", but "starving" can also mean "causing starvation". Changed to `starvationCausingTask` since that seems unambiguous.